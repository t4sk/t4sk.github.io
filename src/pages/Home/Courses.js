import React from "react"
import PropTypes from "prop-types"
import styles from "./Courses.module.css"
import CourseCard from "./CourseCard"

import learnSolidityCardLogo from "../../static/learn-solidity-card.svg"

const COURSES = [
  {
    title: "Learn Solidity",
    logo: learnSolidityCardLogo,
    href: "/courses/learn-solidity",
  },
]

export function Courses(props) {
  return (
    <div className={styles.component}>
      {COURSES.map(course => (
        <CourseCard key={course.title} {...course} />
      ))}
    </div>
  )
}

Courses.propTypes = {}

export default Courses
