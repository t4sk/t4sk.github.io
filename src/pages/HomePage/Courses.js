import React from "react"
import PropTypes from "prop-types"
import styles from "./Courses.module.css"
import CourseCard from "./CourseCard"

import solidityLogo from "../../static/solidity.svg"
import solidityAppLogo from "../../static/solidity-app.svg"

const COURSES = [
  {
    title: "Learn Solidity",
    logo: solidityLogo,
    path: "/courses/learn-solidity",
    // TODO: remove onClick
    onClick: e => {
      e.preventDefault()
      alert("Coming soon...")
    },
  },
  {
    title: "Build a Multi-Sig Wallet in Solidity",
    logo: solidityAppLogo,
    path: "/courses/multi-sig-wallet",
    // TODO: remove onClick
    onClick: e => {},
  },
]

export function Courses(props) {
  return (
    <div className={styles.component}>
      {COURSES.map(course => (
        <div
          key={course.title}
          className={styles.card}
          onClick={course.onClick}
        >
          <CourseCard {...course} />
        </div>
      ))}
    </div>
  )
}

Courses.propTypes = {}

export default Courses
