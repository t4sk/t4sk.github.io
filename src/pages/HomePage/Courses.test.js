import React from "react"
import { shallow } from "enzyme"
import { Courses } from "./Courses"

const props = {}

beforeEach(() => {})

test("it renders", () => {
  const component = shallow(<Courses {...props} />)

  expect(component).toMatchSnapshot()
})
