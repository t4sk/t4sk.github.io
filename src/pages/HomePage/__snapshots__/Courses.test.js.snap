// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="component"
>
  <div
    className="card"
    key="Learn Solidity"
    onClick={[Function]}
  >
    <CourseCard
      logo="solidity.svg"
      onClick={[Function]}
      path="/courses/learn-solidity"
      title="Learn Solidity"
    />
  </div>
  <div
    className="card"
    key="Build a Multi-Sig Wallet in Solidity"
    onClick={[Function]}
  >
    <CourseCard
      logo="solidity-app.svg"
      onClick={[Function]}
      path="/courses/multi-sig-wallet"
      title="Build a Multi-Sig Wallet in Solidity"
    />
  </div>
</div>
`;
